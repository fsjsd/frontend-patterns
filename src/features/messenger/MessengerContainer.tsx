import React, { Suspense } from 'react'
import ContentHeader, { ContentHeaderLabel } from '../../ux/ContentHeader'
import { ContentWrapper } from '../../ux/ContentWrapper'
import Loading from '../../ux/Loading'
const Messenger = React.lazy(() => import('./Messenger'))

const MessengerContainer = () => {
  return (<>
    <ContentHeader>
      <ContentHeaderLabel>Messenger</ContentHeaderLabel>
    </ContentHeader>
    <ContentWrapper>
      <Suspense fallback={<Loading />}>
        <div role="main">
          <Messenger />
        </div>
      </Suspense>
    </ContentWrapper>
  </>
  )
}

export default MessengerContainer