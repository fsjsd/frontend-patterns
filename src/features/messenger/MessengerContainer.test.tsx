import React from 'react'
import MessengerContainer from "./MessengerContainer";
import { render, waitFor } from '@testing-library/react'

describe("Messenger", () => {
  it("renders and matches snapshot", async () => {
    const { container, queryByRole } = render(<MessengerContainer />)
    await waitFor(() => expect(queryByRole("main")).toBeInTheDocument());
    expect(container).toMatchSnapshot();
  })
})