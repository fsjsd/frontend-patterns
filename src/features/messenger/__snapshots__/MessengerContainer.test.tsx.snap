// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Messenger renders and matches snapshot 1`] = `
<div>
  <div
    class="sc-bczRLJ dtMXGe"
  >
    <div
      class="sc-gsnTZi dkPlsI"
    >
      Messenger
    </div>
  </div>
  <div
    class="sc-dkzDqf eGEKCX"
  >
    <div
      role="main"
      style=""
    >
      <div
        class="sc-eCYdqJ jGnNot"
      >
        Coming soon
      </div>
    </div>
  </div>
</div>
`;
